Bug Fixes

KillerOfPie opened the pull request:
- #69  Switched to getTargetBlockExact from getTargetBlock
- #68  Removed available block list, used instanceof Container and made isInventory accept Block obj instead of Material obj
  New valid named for shop blocks in config: Barrel, BlastFurnace, BrewingStand, Chest, Dispenser, Dropper, Furnace, Hopper, ShulkerBox, Smoker
- Removed unnecessary stored objects and some places where code was being used instead of an existing method with the same function.

KillerOfPie committed:
Bug Fixes - Untested
- #69 Switched to getTargetBlockExact from getTargetBlock
- #68 Removed available block list, used instanceof Container and made isInventory accept Block obj instead of Material obj
  New valid named for shop blocks in config: Barrel, BlastFurnace, BrewingStand, Chest, Dispenser, Dropper, Furnace, Hopper, ShulkerBox, Smoker
- Removed unnecessary stored objects and some places where code was being used instead of an existing method with the same function.

SparklingComet commented on src/main/java/org/shanerx/tradeshop/listeners/ShopProtectionListener.java:
```diff
@@ -73,12 +72,14 @@ public ShopProtectionListener(TradeShop instance) {
 		}
 
-		Nameable fromContainer = (Nameable) event.getSource().getLocation().getBlock().getState();
+        Nameable fromContainer = (Nameable) invBlock;
```
Are you sure it is safe to case Nameable to Block? Thought we needed BlockState for that or Container... could be wrong though

KillerOfPie replied:
We do, i goofed when i made that an object and got rid of the getstate, good catch

SparklingComet commented on src/main/java/org/shanerx/tradeshop/objects/ListManager.java:
```diff
@@ -70,14 +72,17 @@ public boolean isDirection(BlockFace face) {
+        if (!(block instanceof Container))
+            return false;
+
+        return inventories.contains(block.getClass().getSimpleName().replaceAll("[\\[\\]]", "").toLowerCase());
```
Uhm... what is going on here? Why are we switching to reflection?

KillerOfPie replied:
So that we dont need to have a list we physically have to update with each version (like how the plugin works for 1.14 but not for things like barrels which had been added but we didnt add it to our internal list)

SparklingComet replied:
Still, we shouldn’t switch to reflection. I don’t think it is actually necessary, it is bothersome to add stuff to the list, but it would still be better

KillerOfPie replied:
Would reflection only on plugin init be better?

SparklingComet replied:
I suppose it would be better...
It would still be far from ideal though. We could keep it as a temporary solution, but we should change it in the future I think

KillerOfPie replied:
Agreed, will make the change later this week and keep an eye out for a better way to generate the list.

KillerOfPie replied:
This is no longer using reflection

SparklingComet reviewed (dismissed):
I don’t like the use of reflection. Isn’t there a better way to achieve this?

KillerOfPie commented:
As far as the chest naming, not that I know of, besides we'll be getting rid of that in the next couple versions

As for the instanceof, not that I could find. if you know of a better way to create a dynamic list of viable Container blocks that we wouldn't need to update whenever minecraft releases a new version be my guest

KillerOfPie committed:
Fixed missing getState() Call

SparklingComet commented:
> As for the instanceof, not that I could find. if you know of a better way to create a dynamic list of viable Container blocks that we wouldn't need to update whenever minecraft releases a new version be my guest

The point is, firstly reflection should not be switched to lightly, secondly we _should_ update for new releases to fix bugs

KillerOfPie commented:
the inventory list is configurable, this is for the internal list that simply checks to make sure that the list items in the list are blocks have inventories.

I will try to find another solution, as even if we are updating for every release an adaptable internal list would still be better since we don't know which version the plugin will actually be running on and we don't want items in the list that are not in a version

KillerOfPie committed:
Fixes

- Added ShopStorage object and enum that is held by the TradeShop object (maybe add this and ShopSign to ListManager later)

TradeShop
- Moved ShopSign and ShopStorage object creation to be earlier in startup

Utils
- Added log method(will replace this and debug with single method that has a log level param that will be in settings)

List Manager
- Added Util Extension
- Removed reflection from storage checking
- Added addOnMats list to have additions from other plugins adding costs and such add onto
- gameMats list gets all game mats and add-on mats

SparklingComet committed:
Remove instances of old TSAF design (custom subcmds)

KillerOfPie commented on src/main/java/org/shanerx/tradeshop/commands/CommandCaller.java:
```diff
@@ -180,19 +160,4 @@ public boolean checkPerm() {
-     * @param exec Command attempting to be run
-     * @return true if permission is NONE or sender has permission
-	 */
-	public boolean checkPerm(TradeCommand exec) {
```
I think this being removed could cause a problem, does it show the checkperm usage on line 64 as having a reference?

SparklingComet replied:
This method was originally part of the custom command part of the tsaf. The class “TradeCommand” was removed in my last commit, since it was meant to represent custom subcommands for /tradeshop
Either way, my IDE did a quick search and all the instances of it being called I removed

KillerOfPie reviewed (commented):
Other than that, looks good. will test later today
